keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Enhancing Network Performance Benchmarking Testcases
+authors:
+  - "@hanamantagoud.v.kandagal@est.tech"
+owning-sig: sig-scalability
+participating-sigs:
+- sig-networks
+- sig-testing
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: 
+creation-date: 2020-08-24
+last-updated: 2020-08-25
+status: implemented
+---
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-goals](#non-goals)
+  - [Challenges and Open Questions [optional]](#challenges-and-open-questions-optional)
+- [Proposal](#proposal)
+  - [Test Scenarios](#test-scenarios)
+  - [Tools](#tools)
+  - [Execution Environment](#execution-environment)
+  - [Future Enhancements](#future-enhancements)
+  - [Benefits](#benefits)
+- [Graduation Criteria](#graduation-criteria)
+  - [Beta -&gt; GA](#beta---ga)
+  - [GA -&gt; Future](#ga---future)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Summary
+
+We are proposing to add more testcases for network performance benchmarking.
+
+## Motivation
+
+As Kubernetes is increasingly being adopted by various telco cloud providers for deployments, 
+it brings additional networking requirements/scenarios to be supported.  
+With wide array of network functions being containerized, the need to analyse and 
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 2 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
+
+b) netperf (https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf) : This framework 
+   measures only the throughput between client-server pods for various MSS(Maximum Segment size). 
+   Other performance metrics such as latency , loss-rate , response time are not measured.
+
+### Goals
+
+The primary goal of this proposal is to enhance the testcases to cover various 
+networking scenarios :
+
+a)  IPv4 , IPv6 , Dual-Stack.
+b)  Protocols: TCP, UDP, HTTP/HTTPS.
+c)  With services such as ClusterIP , NodePort etc.
+
+We intend to submit tests to measure the following network performance metrics for various 
+scenarios such as different services (ClusterIP/NodePort), with IPv4/IPv6 configured etc.
+
+a)  Throughput (For TCP traffic)
```

Comment:
we should have packet rate too, TCP throughput is "easy to manipulate" to have nice metrics (increase MTU, offloading)

Reply:
Yes we can have TCP throughput measurement for various MTU values. Already netperf framework has such testcase for different MSS size.

Reply:
Packet Rate (packet per second) itself is not provided by tools like iperf. So we have to calculate ourselves.

Reply:
I would like to differentiate between packet rate and throughput, is common to use it interchangeable in "marketing" papers, but for performance testing we need to be specific. San we add another point with this metric?
```
b) Packet per seconds
```
One interesting thing you use to see in Linux systems is that you can have maximum throughput only using offloading and jumbo from, because you've reached the packet per second  limit and the only way to get the maximum is using bigger packets

![image](https://user-images.githubusercontent.com/6450081/92213106-4b59c900-ee93-11ea-830a-8b92589b555b.png)